### PR TITLE
feat(trainer): add ForwardKLLoss and ForwardKLWithChunkedOutputLoss to Loss enum

### PR DESCRIPTION
**Pull request by krishdef7**
## Summary

Completes the `Loss` enum by adding the two remaining loss functions from `torchtune.modules.loss`, resolving the TODO left by @Electronic-Waste.

## Changes

- `Loss.ForwardKLLoss`: forward KL divergence, used for knowledge distillation
- `Loss.ForwardKLWithChunkedOutputLoss`: memory-efficient variant via chunked output
- Tests verifying each enum value maps to the correct `torchtune.modules.loss` path

## Notes

`SimPOLoss` was intentionally excluded, it lives in `torchtune.rlhf.loss`
(not `torchtune.modules.loss`) and is marked deprecated upstream.

## References

- TorchTune loss API: https://pytorch.org/torchtune/main/api_ref_loss.html
- Tracking issue: kubeflow/trainer#2839

### CODE REVIEW

File: kubeflow/trainer/types/types_test.py [hunk 1]
```diff
@@ -113,3 +113,86 @@ def test_data_cache_initializer(test_case: TestCase):
         assert test_case.expected_status == FAILED
         assert type(e) is test_case.expected_error
     print("test execution complete")
+<<<<<<< Updated upstream
+=======
```
[the remark this answers is not in the document]
Comment (krishdef7) [reply kubeflow/trainer/types/types_test.py]: Fixed in the latest push, conflict markers and unrelated test functions have been removed.

File: kubeflow/trainer/types/types_test.py [hunk 2]
```diff
@@ -113,3 +113,86 @@ def test_data_cache_initializer(test_case: TestCase):
...
+        result = template.optimize(search_space=search_space)
+
+        mock_client.optimize.assert_called_once_with(
+            trial_template=template,
+            trial_config=None,
+            search_space=search_space,
+            objectives=None,
+            algorithm=None,
+        )
+        assert result == "test-experiment-name"
```
[the remark this answers is not in the document]
Comment (krishdef7) [reply kubeflow/trainer/types/types_test.py]: Fixed in the latest push, conflict markers and unrelated test functions have been removed.

File: kubeflow/trainer/types/types_test.py [hunk 3]
```diff
@@ -113,3 +113,86 @@ def test_data_cache_initializer(test_case: TestCase):
...
+        ),
+    ],
+)
+def test_loss_enum(test_case: TestCase):
+    """Test Loss enum values match TorchTune module paths."""
+    print("Executing test:", test_case.name)
+    loss = test_case.config["loss"]
+    assert loss.value == test_case.expected_output
+    print("test execution complete")
+>>>>>>> Stashed changes
```
[the remark this answers is not in the document]
Comment (krishdef7) [reply kubeflow/trainer/types/types_test.py]: Fixed in the latest push, conflict markers and unrelated test functions have been removed.
